jcstress-core/src/main/java/org/openjdk/jcstress/Main.java

```diff
@@ -51,7 +55,21 @@ public static void main(String[] args) throws Exception {
 
         JCStress jcstress = new JCStress(opts);
         if (opts.shouldList()) {
-            for (String test : jcstress.getTests()) {
+            JCStress.ConfigsWithScheduler configsWithScheduler = jcstress.getConfigs();
+            Set<String> testsToPrint = new TreeSet<>();
+            for (TestConfig test : configsWithScheduler.configs) {
+                if (opts.verbosity().printAllTests()) {
+                    testsToPrint.add(test.toDetailedTest());
+                } else {
+                    testsToPrint.add(test.name);
+                }
+            }
+            if (opts.verbosity().printAllTests()) {
+                System.out.println("All matching tests combinations - " + testsToPrint.size());
+            } else {
+                System.out.println("All matching tests - " + testsToPrint.size());
+            }
+            for (String test : testsToPrint) {
                 System.out.println(test);
             }
```

Comment:
maybe move this code to a separate method, for better readability and maintainability

Reply:
Sure! Done.